### ISSUE TITLE
Epson printer emulation issues

### ISSUE THREAD

**Issue by rderooy** [joncampbell123/dosbox-x#3164]
### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

### Have you checked that no other similar bug report(s) already exists?

- [X] I have searched and didn't find any similar issues.

### What operating system(s) this bug have occurred on?

Linux F35

### What version(s) of DOSBox-X have this bug?

0.83.21git

### Describe the bug

The integrated Epson printer emulation (ESC/P and ESC/P2) has a couple of issues that I ran into in writing my printing guide:

## Graphics printing with Mono-space font
The monospace font has weird spacing issues (printed from Notepad in Windows 3.0)
![Screenshot from 2021-12-21 13-05-28](https://user-images.githubusercontent.com/17701098/146927394-def38675-fde9-4ecb-a671-c22fd810e070.png)

Compare this to the monospace font from the printers "integrated" fonts:

![Screenshot from 2021-12-21 13-26-28](https://user-images.githubusercontent.com/17701098/146929820-9674dad5-0bbc-4be0-a3e4-23e91fe32d50.png)

## Graphics printing with Proportional font
This does not have the weird spacing issues, but the quality is quite low (printed from Write in Windows 3.0)

![Screenshot from 2021-12-21 13-05-55](https://user-images.githubusercontent.com/17701098/146927545-ac0721cc-0b4f-40b2-8646-871f2872d3b8.png)

## Graphics resolution issues
When printing with a graphics resolution other then 180x180 results in corruption. Here is an example of the same text, printed with 360x180:
![Screenshot from 2021-12-21 13-14-13](https://user-images.githubusercontent.com/17701098/146928285-89fd73aa-d143-40ff-8bb6-53d20fa91a46.png)

Likewise printing with 360x360 (supported by some later model Epson dot-matrix printers) also leads to corruption.

## No colour
Some Epson dot-matrix printers support colour like the LQ-860/860+/2500/2550 models. But even with such a model selected, and colour output configured the output is always black-and-white.

### Steps to reproduce the behaviour

see above

### Used configuration

```ini
[parallel]
parallel1=printer

[printer]
printer=true
width=82
height=116
printoutput=ps
multipage=true
timeout=2000
```

**Comment by Wengier** [comment]
I wonder exactly which mono-space font you used for printing from Windows 3.0 Notepad? I tried to print from Windows 3.1 Notepad, one with ``fonts\\courier.ttf`` and one without, using the "generic/text only" option, and got the following results:

![image](https://user-images.githubusercontent.com/8216923/147049868-c96dcea6-5052-4c64-ba03-dd445d8c7d1f.png)

![image](https://user-images.githubusercontent.com/8216923/147049887-2db4269f-0b96-4272-8796-7942c2c99a56.png)

I tried ``printoutput=ps`` and ``printoutput=png`` and the results appeared to be the same.

P.S. With the latest code you should be able to change most config options in [printer] section at runtime. Please check it now.

**Comment by rderooy** [comment]
I think it must be "Courier". Notepad does not really tell you which font is used, nor can you change it.

**Comment by Wengier** [comment]
@rderooy Can you check what is the TTF font used by your DOSBox-X build for printing then? For example, did you put some TTF font as “fonts\courier.ttf” so that DOSBox-X will use it for the printing?

**Comment by rderooy** [comment]
I think that if I print in Windows 3.x from Notepad or Word it uses Graphics printing and therefore uses the Windows 3.x fonts, and the fonts configured in DOSBox-X have no bearing.

The second screenshot I posted above was a redirect from the DOS prompt (``type README.TXT > LPT1:``, and that uses text mode printing and then the fonts configured in DOSBox-X do matter.

If instead I configure a PostScript printer in Windows 3.x and print to a file in DOSBox-X, I get very high quality output. But for the wiki I wanted to also document the printing via the emulated Epson printer.

e.g.:
https://github.com/Wengier/dosbox-x-wiki/wiki/Guide%3AInstalling-Windows-3.x#enabling-printing
https://github.com/Wengier/dosbox-x-wiki/wiki/Guide%3AInstalling-Windows-95#enabling-printing

**Comment by Wengier** [comment]
So for testing I should select the Epson LQ printer instead of "Generic/Text Only". By the way, I cannot find the Epson LQ-860 option in my Windows 3.1, but Epson LQ-850 is available.

**Comment by rderooy** [comment]
That is correct, the Windows 3.x guide says to use the "Epson LQ-860", but that is from Windows 3.11. Your saying that is not available in 3.1?

**Comment by Wengier** [comment]
I just saw the updated guide in my previous post. The option Epsoj LQ-860 does not seem to be available in Windows 3.1, at least in my setup.

**Comment by Wengier** [comment]
@rderooy The Windows 3.1 printer list shows separate Epson LQ-850, Epson LQ-950, and Epson LQ-1050 options as in the screenshot below:

![image](https://user-images.githubusercontent.com/8216923/147102721-0c5d1f19-112b-4b98-be10-ded7d08f09e7.png)

**Comment by rderooy** [comment]
Thanks, I have adjusted the instructions
